lib/playlistGenerator/ConcatSource.js

```diff
@@ -55,6 +55,12 @@ function ConcatSource(loggerInfo, playlistObj, parentObj, track, chunkPath) {
 
 }
 
+function retractBasePath(path) {
```

Comment:
please check PlaylistGenerator.checkFileExists and PlaylistGenerator.handleClipError are not broken
generally, track all usages of path.basename related to chunk path under playlistGenerator dir and see if breaks existing code. maybe you'd better move  retractBasePath to persistenceFormat.js to be globally accessible...